tests/test_admin.py

```diff
@@ -32,3 +39,75 @@ def test_oas_fields_disabled(admin_client: Client, settings):
 
     assert "oas" not in form.fields
     assert "oas_file" not in form.fields
+
+
+def test_get_connection_check_correct_status_code(admin_client: Client, settings):
+    service = ServiceFactory.create(
+        api_root="https://example.com/",
+        api_connection_check_path="foo",
+        auth_type=AuthTypes.zgw,
+        client_id="my-client-id",
+        secret="my-secret",
+    )
+    with requests_mock.Mocker() as m:
+        m.get("https://example.com/foo", status_code=401)
+        url = reverse(
+            "admin:zgw_consumers_service_change", kwargs={"object_id": service.id}
+        )
+        response = admin_client.get(url)
+
+        connection_check_inner_html = '<label>Connection check status code:</label><div class="readonly">401</div>'
+        assertInHTML(
+            connection_check_inner_html, response.render().content.decode("utf-8")
+        )
```

Comment:
```suggestion
        assertContains(response, connection_check_inner_html, html=True)
```
much more elegant :)

if you need the raw bytes of a response, you can also do `response.content` instead of going through `response.render().content`, but here especially the `.decode` and str vs bytes indicated that _there must be a better way_